packages/gitmoji-changelog-core/src/index.js

```diff
@@ -120,11 +135,17 @@ async function generateChangelog(from, to, {
   if (from === TAIL) {
     tagsToProcess = [...tagsToProcess, TAIL]
   } else {
-    const fromIndex = tagsToProcess.findIndex(tag => semver.eq(tag, from))
-    tagsToProcess.splice(fromIndex + 1)
+    try {
+      const fromIndex = tagsToProcess.findIndex(tag => semver.eq(tag, from))
+      tagsToProcess.splice(fromIndex + 1)
 
-    if (hasNext && isEmpty(tagsToProcess)) {
-      tagsToProcess.push(HEAD)
+      if (hasNext && isEmpty(tagsToProcess)) {
+        tagsToProcess.push(HEAD)
+      }
+    } catch (e) {
+      if (e.name === 'TypeError') {
+        throw FunctionalError(e.message)
```

Comment:
It's hard to know what kind of functional error it is. I think you should write a specific error message to  explain the error. It will help to understand the code and handled errors.

Reply:
These are errors from semver which are very specific and explicit.